docs/evm/near-web3-provider.md

```diff
@@ -0,0 +1,182 @@
+---
+id: near-web3-provider
+title: NEAR web3 provider
+sidebar_label: Web3 provider
+---
+
+The [near-web3-provider](https://github.com/near/near-web3-provider) is a custom provider that can be set when working in a web3 environment. There are two typical use cases for a web3 provider:
+
+1. [Frontend](#frontend) — interacting with smart contracts via a user interface on the web
+2. [Backend](#backend) — testing and deploying EVM-based contracts
+
+## Frontend
+
+The [NPM package](https://npmjs.com/package/near-web3-provider) and can be added to a NodeJS project with:
+
+    npm install near-web3-provider --save
+    
+Or, to follow the snippets below, you may want to use:
+
+    npm install near-web3-provider web3-eth-contract --save
+
+In the frontend JavaScript the provider can be instantiated with its constructor. There are two modes of a `NearProvider` class:
+
+1. Read-only — can only access `view` calls to the NEAR EVM.
+2. Signed — has access to a NEAR account and can interact with `view` and `call` methods.
+
+**Note**: in Ethereum, executing a read-only method is referred to as `call`, while a state-changing method is referred to as `send`. In NEAR, it's `view` and `call` and this terminology will be used widely in the documentation.
+
+### Instantiating read-only
+
+```js
+const { NearProvider } = require('near-web3-provider');
+// …
+const nearProvider = new NearProvider({
+  networkId: this.nearNetwork,
+  keyStore: new nearAPI.keyStores.BrowserLocalStorageKeyStore(),
+  isReadOnly: true
+});
+```
+
+###  Instantiating signed
+
+For the signed instance, this assumes that the browser has local storage containing a key to the account specified.
+
+For folks unfamiliar with NEAR Wallet, a brief recap as the workflow in NEAR is different than that of a browser extension or by scanning QR codes. Let's imagine a user wants to use a NEAR betanet account to interact with the EVM. The first step is to create an account with Wallet. (Non-mainnet networks allow for free account creation with an initial balancer of Ⓝ.)
+
+NEAR Betanet Wallet:
+
+https://wallet.betanet.near.org
+
+At this point, the NEAR Wallet website's local storage contains a key for this account. When you want to interact with a frontend using your account, you'll typically have the frontend redirect to Wallet, requesting the creation a special function-call access key. After allowing the creation, the browser redirects back to the dApp frontend where local storage contains this new key.
```

Comment:
"where local storage contains this new key" => "with local storage containing this new key"